--- a/setup.py
+++ b/setup.py
@@ -144,6 +144,10 @@ class cmake_build_ext(build_ext):
             # Default build tool to whatever cmake picks.
             build_tool = []
 
+        cmake_args += [
+            '-DTORCH_CUDA_ARCH_LIST=6.0 6.1 7.0 7.5 8.0 8.6 8.9 9.0+PTX',
+        ]
+
         subprocess.check_call(
             ['cmake', ext.cmake_lists_dir, *build_tool, *cmake_args],
             cwd=self.build_temp)
