--- a/launch.py
+++ b/launch.py
@@ -7,6 +7,7 @@ import shlex
 import subprocess
 from functools import lru_cache
 import installer
+from nvidia_pstate import set_pstate_high, set_pstate_low
 
 
 debug_install = installer.log.debug if os.environ.get('SD_INSTALL_DEBUG', None) is not None else lambda *args, **kwargs: None
@@ -262,4 +263,5 @@ def main():
 
 
 if __name__ == "__main__":
+    set_pstate_low()
     main()
--- a/modules/shared_state.py
+++ b/modules/shared_state.py
@@ -3,6 +3,7 @@ import sys
 import time
 import datetime
 from modules.errors import log
+from nvidia_pstate import set_pstate_high, set_pstate_low
 
 
 class State:
@@ -83,6 +84,8 @@ class State:
         self.time_start = time.time()
         if self.debug_output:
             log.debug(f'State begin: {self.job}')
+        if self.job != "load":
+            set_pstate_high()
         modules.devices.torch_gc()
 
     def end(self, api=None):
@@ -92,6 +95,8 @@ class State:
             self.time_start = time.time()
         if self.debug_output:
             log.debug(f'State end: {self.job} time={time.time() - self.time_start:.2f}')
+        if self.job != "load":
+            set_pstate_low()
         self.job = ""
         self.job_count = 0
         self.job_no = 0
