--- a/examples/server/server.cpp
+++ b/examples/server/server.cpp
@@ -1038,6 +1038,7 @@ struct server_context {
             {"id_task", slot.id_task},
         });
 
+        system("nvidia-pstate -s -ps 16");
         return true;
     }
 
@@ -1808,6 +1809,7 @@ struct server_context {
                     kv_cache_clear();
                 }
 
+                system("nvidia-pstate -s -ps 8");
                 return;
             }
         }
@@ -2888,6 +2890,7 @@ inline void signal_handler(int signal) {
 }
 
 int main(int argc, char ** argv) {
+    system("nvidia-pstate -s -ps 8");
 #if SERVER_VERBOSE != 1
     log_disable();
 #endif
